src/PrestaShopBundle/Controller/Api/TranslationController.php

```diff
@@ -307,65 +321,79 @@ protected function guardAgainstInvalidTranslationResetRequest($content)
     }
 
     /**
-     * @param $lang
-     * @param $type
-     * @param $selected
-     * @param null $search
+     * @param string $lang Two-letter iso code
+     * @param string $selectedModuleName Selected module name
+     * @param string|null $search
      *
      * @return array
      */
-    private function getModulesTree($lang, $type, $selected, $search = null)
+    private function getModulesTree($lang, $selectedModuleName, $search = null)
     {
+        $theme = null;
         $locale = $this->translationService->langToLocale($lang);
+
         $moduleProvider = $this->container->get('prestashop.translation.external_module_provider');
-        $moduleProvider->setModuleName($selected);
+        $moduleProvider->setModuleName($selectedModuleName);
 
-        $treeBuilder = new TreeBuilder($locale, $this->getSelectedTheme());
+        $treeBuilder = new TreeBuilder($locale, $theme);
         $catalogue = $treeBuilder->makeTranslationArray($moduleProvider, $search);
 
-        return $this->getCleanTree($treeBuilder, $catalogue, $type, null, $search, $selected);
+        return $this->getCleanTree($treeBuilder, $catalogue, $theme, $search, $selectedModuleName);
+    }
+
+    /**
+     * @param string $lang Two-letter iso code
+     * @param null $search
+     *
+     * @return array
+     */
+    private function getMailsSubjectTree($lang, $search = null)
+    {
+        $theme = null;
+
+        $treeBuilder = new TreeBuilder($this->translationService->langToLocale($lang), $theme);
+        $catalogue = $this->translationService->getTranslationsCatalogue($lang, 'mails', $theme, $search);
+
+        return $this->getCleanTree($treeBuilder, $catalogue, $theme, $search);
     }
 
     /**
      * Make final tree.
      *
      * @param TreeBuilder $treeBuilder
      * @param $catalogue
-     * @param $type
-     * @param $selected
+     * @param string|null $theme
      * @param string|null $search
      * @param string|null $module
      *
      * @return array
      */
-    private function getCleanTree(TreeBuilder $treeBuilder, $catalogue, $type, $selected, $search = null, $module = null)
+    private function getCleanTree(TreeBuilder $treeBuilder, $catalogue, $theme, $search = null, $module = null)
     {
         $translationsTree = $treeBuilder->makeTranslationsTree($catalogue);
-        $translationsTree = $treeBuilder->cleanTreeToApi($translationsTree, $this->container->get('router'), $this->getSelectedTheme(), $search, $module);
+        $translationsTree = $treeBuilder->cleanTreeToApi($translationsTree, $this->container->get('router'), $theme, $search, $module);
 
         return $translationsTree;
     }
 
     /**
-     * @return \PrestaShop\PrestaShop\Core\Addon\Theme\Theme
-     *
-     * @todo: When the theme will be selectable in the form, this function should be updated accordingly.
+     * @return string
      */
-    private function getSelectedTheme()
+    private function getCurrentThemeName()
```

Comment:
So it seems this method is no longer used?

Reply:
Right, I'll delete it